### PR TITLE
Fix scheme required for webhook url in amtool

### PR DESCRIPTION
**Pull request by grobinson-grafana**
This commit fixes issue #3505 where amtool would fail with `error: scheme required for webhook url` when using amtool with `--alertmanager.url`.

The issue here is that `UnmarshalYaml` for `WebhookConfig` checks if the scheme is present when `c.URL` is non-nil. However, `UnmarshalYam`l for `SecretURL` returns a non-nil, default value `url.URL{}` if the response from `api/v2/status` contains `<secret>` as the webhook URL.

### CODE REVIEW

File: test/cli/acceptance/cli_test.go [hunk 1]
```diff
@@ -168,3 +168,36 @@ receivers:
 		t.Errorf("Incorrect number of silences queried, expected: %v, actual: %v", expectedSils, len(sils))
 	}
 }
+
+func TestRoutesShow(t *testing.T) {
```
Comment (grobinson-grafana) [inline test/cli/acceptance/cli_test.go:172]: Revert the fix and this test fails:

```
--- FAIL: TestRoutesShow (1.09s)
    cli_test.go:202: amtool: error: scheme required for webhook url

    cli_test.go:203:
        	Error Trace:	/Users/grobinson/go/src/github.com/prometheus/alertmanager/test/cli/acceptance/cli_test.go:203
        	Error:      	Received unexpected error:
        	            	exit status 1
        	Test:       	TestRoutesShow
FAIL
```

File: config/notifiers.go [hunk 2]
```diff
@@ -503,11 +503,6 @@ func (c *WebhookConfig) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	if c.URL != nil && c.URLFile != "" {
 		return fmt.Errorf("at most one of url & url_file must be configured")
 	}
-	if c.URL != nil {
```
Comment (grobinson-grafana) [inline config/notifiers.go:506]: We think this is duplicate code as `SecretURL` is just a `URL`, whose `UnmarshalYaml` function also calls `parseURL` which contains the same check.

**Comment by simonpasquier** [comment]
Thanks! I'd be inclined to have the fix in the release-0.26 branch.

**Comment by grobinson-grafana** [comment]
Apologies for the force push, there was a spelling mistake in the commit message.

File: test/cli/acceptance.go [hunk 3]
```diff
@@ -658,6 +658,17 @@ func (am *Alertmanager) UpdateConfig(conf string) {
...
 }
 
+func (am *Alertmanager) ShowRoute() ([]byte, error) {
+	return am.showRouteCommand()
+}
+
+func (am *Alertmanager) showRouteCommand() ([]byte, error) {
+	amURLFlag := "--alertmanager.url=" + am.getURL("/")
+	args := []string{amURLFlag, "config", "routes", "show"}
+	cmd := exec.Command(amtool, args...)
```
Comment (gotjosh) [inline test/cli/acceptance.go:668]: Can we also add one for `config routes test`?

**Review by gotjosh — APPROVED** [review]
LGTM!

Thank you very much for your contribution.

**Pull request merged**